Robot/neopixel-leds/src/test/java/org/usfirst/frc/team4999/tools/BufferDisplay.java

```diff
@@ -0,0 +1,175 @@
+            return;
+        }
+        int[] payload = new int[rawPacketBytes.length - 1];
+        for(int i = 0; i < payload.length; i++) {
+            payload[i] = unsignedByteValue(rawPacketBytes[i + 1]);
+        }
+        
+        int len = payload.length;
+        int command = payload[0];
+        int address = payload[1];
+        Color rgb = new Color(payload[2], payload[3], payload[4]);
+        switch(command) {
+            case 1:
+                if(len == 1) {
+                    notifyBufferListeners();
+                } else {
+                    System.err.format("Invalid display packet: expected length 1, actual length %d\n", len);
+                }
+                break;
+            case 2:
+                if(len == 5) {
+                    paintPattern(address, rgb, 1, 0, 0);
+                } else {
+                    System.err.format("Invalid single packet: expected length 5, actual length %d\n", len);
+                }
+                break;
+            case 3:
+                if(len == 6) {
+                    int count = payload[5];
+                    paintPattern(address, rgb, count, 0, 0);
+                } else {
+                    System.err.format("Invalid run packet: expected length 5, actual length %d\n", len);
+                }
+                break;
+            case 4:
+                if(len == 7) {
+                    int count = payload[5];
+                    int stride = payload[6];
+                    paintPattern(address, rgb, count, stride, 0);
+                } else {
+                    System.err.format("Invalid stride packet: expected length 7, actual length %d\n", len);
+                }
+                break;
+            case 5:
+                if(len == 8) {
+                    int count = payload[5];
+                    int stride = payload[6];
+                    int maxlength = payload[7];
+                    paintPattern(address, rgb, count, stride, address + maxlength);
+                } else {
+                    System.err.format("Invalid truncated stride packet: expected length 8, actual length %d\n", len);
+                }
+                break;
+            default:
+                System.err.format("Invalid packet: unknown command 0x%x\n", command);
+        }
+    }
+
+    private void paintPattern(int address, Color rgb, int count, int stride, int limit) {
+        if(limit == 0 || limit > buffer.length) {
+            limit = buffer.length;
+        }
+        do {
+            for(int i = 0; i < count; i++) {
+                int pixel = address + i;
+                if(pixel < limit) {
+                    buffer[pixel] = rgb;
+                }
+            }
+            address += stride;
+        } while (address < limit && stride > 0);
+    }
+
+    @Override
+    public void show(Packet[] commands) {
+        for(Packet packet : commands) {
```

Comment:
Show is called by the Animator on every frame.
The code to send show packets resides in the NeoPixels display class, not in the animator, so this display should never receive a show packet. Calling updateBufferListeners() after parsing all packets is a workaround the fact that no show packet will ever be received.
I do think it would be better to move the logic for sending show packets into the Animator, however that would involve changing the actual library code. Since I was trying to limit the scope of this pull request to just test code, and not actual library code, I elected to use the workaround instead.

Reply:
I'm ok to leave the redundant call for now, but you should consider completing the rework to make it consistent and correct.  Perhaps as a phase II of this effort.